src/components/home/Popup.tsx

```diff
@@ -0,0 +1,114 @@
+/** @jsx jsx */
+import React from 'react';
+import { jsx, SxStyleProp } from 'theme-ui';
+
+import { IPost } from '../../utils/interface';
+import { theme } from '../../utils/theme';
+
+export interface PopupProps {
+  /** The post details. */
+  post: IPost;
+  /** The callable that is called once the close button is pressed. */
+  closeHandler: () => void;
+}
+
+/**
+ * A component representing a popup for a post.
+ * 
+ * This popup will occur once a post is clicked, and contains the full
+ * message, the author's name, and the picture in its original
+ * aspect ratio.
+ * 
+ * Desktop view:
+ *  - Spans the entire page
+ *  - The image area is about 60% of the width, and the text area is the rest
+ *  - A small header above the image area contains the exit button
+ * 
+ * Mobile view:
+ *  - Only the picture will be shown in mobile view, as the text can be
+ *    easily found by clicking the description.
+ * 
+ * @returns the popup of a post.
+ */
+const Popup: React.FC<PopupProps> = ({post, closeHandler}) => {
+  const {name, message, photo} = post;
+
+  const popupDiv: SxStyleProp = {
+    position: 'fixed',
+    display: 'flex',
+    flexDirection: 'row',
+    backgroundColor: 'popup.popupBg',
+    top: '0%',
+    zIndex: 1000,
+  }
+
+  const photoRegionDiv: SxStyleProp = {
+    height: '100vh',
+    display: 'flex',
+    flexDirection: 'column',
+    width: ['100vw', '65vw'],
+    ml: 0,
+    mr: 'auto',
+  }
+
+  const textRegionDiv: SxStyleProp = {
+    height: '100vh',
+    width: '35vw',
+    mr: 0,
+    ml: 'auto',
+    px: '2%',
+    py: '2%',
+    backgroundColor: 'popup.textBg',
+    visibility: ['collapse', 'visible']
+  }
+
+  const authorTextStyle: SxStyleProp = {
+    fontSize: 'fontSizes.medium',
+    fontFamily: 'fonts.body',
+  }
+
+  const lineStyle: SxStyleProp = {
+    border: '1px solid gray',
+    height: '2px',
+    width: '100%',
+    margin: 'auto',
+    mb: '7%'
+  }
+
+  const borderStyle: SxStyleProp = {
+    backgroundColor: 'primary',
+    width: '100%',
+    height: '7%',
+  }
+
+  const photoContainerStyle: SxStyleProp = {
+    height: `${100 - parseInt(borderStyle.height.toString()) * 2}%`,
+    width: '100%',
+    textAlign: 'center',
+  }
+
+  const photoStyle: SxStyleProp = {
+    objectFit: 'contain',
+    height: '100%',
+    width: 'auto',
+  }
+
+  return (
+    <div sx={popupDiv}>
+      <div sx={photoRegionDiv}>
+        <div sx={borderStyle} />
```

Comment:
The photo region has a border? Or are you referring to the background of the photo region that's dark blue.

Reply:
If so, rather than having two `div` before and after the image, just have the `photoRegionDiv` to be a flex box with that background colour and then center the image to the middle of the flex box.

Reply:
That is a much better idea gimmie a second

Reply:
Actually, rethinking about it, wouldn't having one flexbox for the entire photo area make `position: absolute` required to position the x button? Either that or you need to define another div for the flexbox above the photo container, which makes you make another div below the photo, which kinda ruins the point.

Also, is it even possible to declare certain parts of a flexbox transparent? If not, then there wouldn't be a transparent view of the images below. I'll keep it for now, and if you can correct me I'll gladly change it.